designer/client/src/components/graph/node-modal/fragment-input-definition/settings/variants/fields/UserDefinedListInput.tsx

```diff
@@ -27,11 +41,47 @@ export const UserDefinedListInput = ({ fixedValuesList, path, onChange, variable
         onChange(`${path}.inputConfig.fixedValuesList`, filteredItemList);
     };
 
+    const temporaryListItemTyp = useMemo(
+        () => ({
+            type: "TypedClass",
+            display: "",
+            refClazzName: typ.refClazzName,
+            params: [],
+        }),
+        [typ.refClazzName],
+    );
+
     const handleAddNewListItem = () => {
-        const isUniqueValue = fixedValuesList.every((fixedValuesItem) => fixedValuesItem.label.trim() !== temporaryListItem.trim());
-        const isEmptyValue = !temporaryListItem;
+        const isUniqueValueValidator = uniqueValueValidator(fixedValuesList.map((fixedValuesList) => fixedValuesList.label));
+        const mandatoryParameterValidator = validators.MandatoryParameterValidator();
+
+        if (!mandatoryParameterValidator.isValid(temporaryListItem)) {
+            setTemporaryValueErrors((prevState) => [
+                ...prevState,
+                {
+                    fieldName: temporaryItemName,
+                    typ: temporaryListItemTyp.refClazzName,
+                    description: mandatoryParameterValidator.description(),
+                    message: mandatoryParameterValidator.message(),
+                },
+            ]);
+            return;
+        }
+
+        if (!isUniqueValueValidator.isValid(temporaryListItem)) {
+            setTemporaryValueErrors((prevState) => [
```

Comment:
Theres a weird edge case:
1. Add some list item "A"
2. Type the "A" and try to add it to the list - the `This field has to be unique` should appear
3. Remove the element from the list
4. Try to add the element - when the value is unchanged, the error will still appear